apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.Snippets/ErrorReportingSnippets.cs

```diff
@@ -0,0 +1,166 @@
+        {
+            _polling = new ErrorEventEntryPolling();
+            _testId = Utils.GetTestId();
+
+            var builder = new WebHostBuilder().UseStartup<ErrorReportingTestApplication>();
+            _server = new TestServer(builder);
+            _client = _server.CreateClient();
+
+            _startTime = DateTime.UtcNow;
+        }
+
+        /// <summary>
+        /// Test for the snippet sample ReportUnhandledExceptions which is a
+        /// confiuration sample. Only way to test that the configuratio is ok
+        /// is to check that logging of errors works as expected.
+        /// </summary>
+        [Fact]
+        public async Task LogsUnhandledException()
+        {
+            await Assert.ThrowsAsync<Exception>(() =>
+                _client.GetAsync($"/Home/ThrowsException/{_testId}"));
+
+            var errorEvents = _polling.GetEvents(_startTime, _testId, 1);
+            Assert.Single(errorEvents);
+            VerifyErrorEvent(errorEvents.First(), _testId, "ThrowsException");
+        }
+
+        /// <summary>
+        /// Checks that an <see cref="ErrorEvent"/> contains valid data.
+        /// </summary>
+        /// <param name="errorEvent">The event to check.</param>
+        /// <param name="testId">The id of the test.</param>
+        /// <param name="functionName">The name of the function the error occurred in.</param>
+        private void VerifyErrorEvent(ErrorEvent errorEvent, string testId, string functionName)
+        {
+            Assert.Equal(ErrorReportingTestApplication.Service, errorEvent.ServiceContext.Service);
+            Assert.Equal(ErrorReportingTestApplication.Version, errorEvent.ServiceContext.Version);
+
+            Assert.Contains(functionName, errorEvent.Message);
+            Assert.Contains(testId, errorEvent.Message);
+
+            Assert.Equal(HttpMethod.Get.Method, errorEvent.Context.HttpRequest.Method);
+            Assert.False(string.IsNullOrWhiteSpace(errorEvent.Context.HttpRequest.Url));
+            Assert.True(errorEvent.Context.HttpRequest.ResponseStatusCode >= 200);
+
+            if (_isWindows)
+            {
+                Assert.False(string.IsNullOrWhiteSpace(errorEvent.Context.ReportLocation.FilePath));
+                Assert.True(errorEvent.Context.ReportLocation.LineNumber > 0);
+            }
+            Assert.Equal(functionName, errorEvent.Context.ReportLocation.FunctionName);
+        }
+
+        public void Dispose()
+        {
+            _client.Dispose();
+            _server.Dispose();
+        }
+    }
+
+    /// <summary>
+    /// A simple web application base to test the <see cref="GoogleExceptionLogger"/>
+    /// and associated classes.
+    /// </summary>
+    [SnippetOutputCollector]
+    internal class ErrorReportingTestApplication
+    {
+        public const string Service = "service-name";
+        public const string Version = "version-id";
+        protected readonly string ProjectId = Utils.GetProjectIdFromEnvironment();
```

Comment:
We should probably move Utils into the common client testing project, and give it a better name :)

Reply:
Agreed, at least part of it should be moved.
Utils is defined on Google.Cloud.Diagnostics.Common.Tests and used throughout diagnostic testing, and there are a method and a class defined there that are diagnostics specific. Moving the other methods in Utils to a better named class in Google.Cloud.ClientTesting makes all the sense int the world but will requiere a little bit of (dumb) work across all the diagnostic projects 
(and solutions) where this might be being used.
I think that it's better if we move that work to another PR? I created issue #2198 so as not to forget.
@jskeet , @iantalarico if you'd like this work done sooner or as part of the current PR let me know.

Reply:
Definitely a separate PR - thanks for raising the issue to keep track.